developer_tools/minprintf.h

```diff
@@ -0,0 +1,336 @@
+/**
+ * @file minprintf.h
+ * @brief Implementation of a function similar to printf.
+ * @details
+ * printf statement rewritten (as minprinf) in c without using stdio.h library.
+ * Syntax of minprintf is same as printf.
+ * Currently minprintf handles:
+ * 	Integers, Doubles, floats, characters, strings.
+ * The format specifiers and escape sequence is same as for printf.
+ * User can also specify the width and precision if requires just like in the case of printf.
+ * How to use it:
+ * 	- First include minprintf.h in your code.
+ * 	- Then type minprintf(), and pass required parameters to it.
+ * 	- As already specified it is similar to prints so use it in the same way.
+ * @author [Jaskarn Singh] (https://github.com/Jaskarn7)
+*/
+
+#include <stdlib.h> // For malloc and free functions.
+#include <unistd.h> // For write function.
+#include <stdarg.h> // For va_start and va_arg functions.
+
+/**
+ * INT_MAX_LENGTH is used as standard length of string to store integers.
+ * PRECISION_FOR_FLOAT is default precision for float or double if not specified.
+*/
+#define INT_MAX_LENGTH 10
+#define PRECISION_FOR_FLOAT 8
+
+
+/**
+ * buffr used to store character in certain situations.
+ * bufsize is 0, if buffr is empty else 1.
+*/
+static char buffr;
+int bufsize = 0;
+
+/**
+ * @param a The power of ten which is to be returned.
+ * @return Ten to the power a. 
+ */
+int power_of_ten(int a)
+{
+	int n = 1; ///< This number will be returned as ten to power of a.
+	for (int i = 1; i <= a; ++i)
+		n *= 10 ;
+	return n;
+}
+
+/**
+ * Checks if a character is a number.
+ * @param c number to be checked.
```

Comment:
Is this a number or a character?
```suggestion
 * @param c number to be checked
```